homeassistant/components/yolink/sensor.py

```diff
@@ -252,6 +253,12 @@ def __init__(
     @callback
     def update_entity_state(self, state: dict) -> None:
         """Update HA Entity State."""
+        if (
+            self.coordinator.device.device_type == ATTR_DEVICE_SIREN
+            and state.get("powerSupply") == "usb"
+        ):
+            # When the power supply mode is usb, the power information is wrong
+            state["battery"] = None
```

Comment:
We shouldn't modify the data coming from the api. If it needs adjustment it should be done in the device library.

Is this a device that can both be powered by USB and battery and power supply can change during the entity lifetime?

Reply:
Thanks, I will change on API library, When the device has a battery installed and is powered by USB, the battery power obtained is inaccurate.

Reply:
I think just remove message convert is fine, When a battery is installed in the device, if device changed to USB power supply, the battery power will be displayed as unavailable, which is unreasonable.

Reply:
Is this the current behavior, that the sensor becomes unavailable if the power supply changes to USB? Where is the logic for this?

Reply:
Setting the value `None` as the sensor native value will be translated to unknown state in the base entity class, not unavailable state.